library/kani/src/invariant.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! This module introduces the `Invariant` trait as well as its implementation
+//! for primitive types.
+
+/// This trait should be used to specify and check type safety invariants for a
+/// type.
+///
+/// Note: Type safety might be checked automatically by Kani in the future.
+pub trait Invariant
```

Comment:
I think it would be great if we can improve this documentation a bit, and maybe include an example. I would also add a link to this definition:
https://rust-lang.github.io/unsafe-code-guidelines/glossary.html#validity-and-safety-invariant

I was also wondering if we should just name this trait `SafetyInvariant`.